SocialCareCaseViewerApi.Tests/V1/UseCase/FormSubmissionUseCaseTests.cs

```diff
@@ -694,6 +663,78 @@ public void ExecuteUpdateSubmissionToApprovedThrowsUpdateSubmissionExceptionIfAp
             act.Should().Throw<UpdateSubmissionException>()
                 .WithMessage($"Worker with email {request.EditedBy} cannot approve the submission as they created the submission");
         }
+
+        [Test]
+        public void ExecuteByQueryThrowsQueryCaseSubmissionsExceptionIfNoParametersFound()
+        {
+            var request = TestHelpers.CreateQueryCaseSubmissions();
+
+            Action act = () => _formSubmissionsUseCase.ExecuteGetByQuery(request);
+
+            act.Should().Throw<QueryCaseSubmissionsException>()
+                .WithMessage("Provide at minimum one query parameter");
+        }
+
+        [Test]
+        public void ExecuteByQueryOnlyGetsFormsWithQueriedFormId()
+        {
+            const string testFormId = "foo";
+            var request = TestHelpers.CreateQueryCaseSubmissions(formId: testFormId);
+
+            var builder = Builders<CaseSubmission>.Filter;
+            var filter = builder.Empty;
+            filter &= Builders<CaseSubmission>.Filter.Eq(s => s.FormId, testFormId);
+
+            _mockMongoGateway.Setup(m =>
+                m.LoadRecordsByFilter(It.IsAny<string>(), It.IsAny<FilterDefinition<CaseSubmission>>()));
+
+            _formSubmissionsUseCase.ExecuteGetByQuery(request, filter);
+
+            _mockMongoGateway.Verify(x =>
+                x.LoadRecordsByFilter(MongoConnectionStrings.Map[Collection.ResidentCaseSubmissions], It.IsAny<FilterDefinition<CaseSubmission>>()), Times.Once);
+        }
+
+        [Test]
+        public void ExecuteByQueryOnlyGetsFormsWithQueriedSubmissionState()
+        {
+            var submissionStates = new List<SubmissionState>() { SubmissionState.InProgress };
+            var submissionStatesRequest = new List<string>() { "in_progress" };
+            var request = TestHelpers.CreateQueryCaseSubmissions(submissionStates: submissionStatesRequest);
+
+            var builder = Builders<CaseSubmission>.Filter;
+            var filter = builder.Empty;
+            filter &= Builders<CaseSubmission>.Filter.In(s => s.SubmissionState, submissionStates);
+
+            _mockMongoGateway.Setup(m =>
+                m.LoadRecordsByFilter(It.IsAny<string>(), It.IsAny<FilterDefinition<CaseSubmission>>()));
+
+            _formSubmissionsUseCase.ExecuteGetByQuery(request, filter);
+
+            _mockMongoGateway.Verify(x =>
+                x.LoadRecordsByFilter(MongoConnectionStrings.Map[Collection.ResidentCaseSubmissions], It.IsAny<FilterDefinition<CaseSubmission>>()), Times.Once);
```

Comment:
Is there a way to improve this test to confirm the function was called with something equivalent to the filter object we constructed?